pkg/azure/client.go

```diff
@@ -0,0 +1,33 @@
+package azure
+
+import (
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/sonobuoy-plugin/v5/pkg/azure/credentials"
+	"github.com/giantswarm/sonobuoy-plugin/v5/pkg/azure/internal/client"
+)
+
+type ClientConfig struct {
+	ServicePrincipal *credentials.ServicePrincipal
+}
+
+func NewClient(config ClientConfig) (*Client, error) {
+	if config.ServicePrincipal == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.ServicePrincipal can't be nil")
+	}
+
+	sp := config.ServicePrincipal
+	creds := auth.NewClientCredentialsConfig(sp.ClientID, sp.ClientSecret, sp.TenantID)
+	authorizer, err := creds.Authorizer()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
```

Comment:
This is not ready for multi tenant service principals, I guess it's fine for now.

Reply:
If this would be changed for multi-tenant SPs, what would need to change? Would the API look a lot different? I'm asking to figure if that refactoring is easily possible in the future or if we should prepare for that already now?

Full disclosure: I'm somewhat testing new patterns / design here to see if we could use something similar in `azure-operator`.